### PR TITLE
add fram_intra_process bool to message_info

### PR DESCRIPTION
**Pull request by wjwwood**
Connects to ros2/demos#12

### CODE REVIEW

File: rmw/include/rmw/types.h [hunk 1]
```diff
@@ -144,6 +144,7 @@ typedef struct RMW_PUBLIC_TYPE rmw_message_info_t
 {
   // const rmw_time_t received_timestamp;
   rmw_gid_t publisher_gid;
+  bool from_intra_process;
```
Comment (jacquelinekay) [inline rmw/include/rmw/types.h]: rmw is supposed to have C headers, so this should be an int, not a bool, right?

Comment (wjwwood) [inline rmw/include/rmw/types.h]: C has bool: http://en.cppreference.com/w/c/types/boolean

I should include `#include <stdbool.h>` though. I'll add that.

**Comment by jacquelinekay** [comment]
+1

**Comment by tfoote** [comment]
+1

**Comment by esteve** [comment]
+1

**Pull request merged**